Collected traffic metrics from requests and responses

g4s8 opened the pull request:
#425 - collected size of requests body publisher and response body publisher and increments traffic metrics.

olegmoz commented on src/main/java/com/artipie/http/TrafficMetricSlice.java:
```diff
@@ -0,0 +1,238 @@
+         */
+        private final Publisher<ByteBuffer> origin;
+
+        /**
+         * Metrics to update.
+         */
+        private final Metrics metrics;
+
+        /**
+         * Metrics name to update.
+         */
+        private final String name;
```
@g4s8 you may pass `Counter` to this class instead of two fields

olegmoz commented on src/test/java/com/artipie/http/TrafficMetricSliceTest.java:
```diff
@@ -0,0 +1,108 @@
+
+                @Override
+                public void gauges(final Map<String, Long> data) {
+                    if (data.containsKey(TrafficMetricSliceTest.KEY_REQ)) {
+                        request.set(data.get(TrafficMetricSliceTest.KEY_REQ));
+                    }
+                    if (data.containsKey(TrafficMetricSliceTest.KEY_RSP)) {
+                        response.set(data.get(TrafficMetricSliceTest.KEY_RSP));
+                    }
```
@g4s8 it seems that this code block is not needed and so may be removed. It would be even better to throw an exception here, because we do not expect the slice to update any gauges.

olegmoz commented on src/test/java/com/artipie/http/TrafficMetricSliceTest.java:
```diff
@@ -0,0 +1,108 @@
+            .response(
+                new RequestLine(RqMethod.GET, "/foo").toString(),
+                Collections.emptyList(),
+                Flowable.just(ByteBuffer.wrap(new byte[size]))
```
@g4s8 I'd suggest to have different size for request and response, otherwise we may mix up response and request in metrics and never know

g4s8 commented:
@olegmoz thanks, I've fixed all comment. Also, I extracted `TestMetricsOutput` from another test class to use it instead of anonymous class for output.

0crat commented:
Job `gh:artipie/artipie#426` is not assigned, can't get performer

<!-- https://www.0crat.com/footprint/CT2E6TK9B/cd694062-ab70-425f-9547-124e5f74353e, version: 0.54.5, hash: ${buildNumber} -->
